Automattic/wp-calypso
# Courses: Refactor to follow convention

There is a strong convention in Calypso to wrap components in the file that defines the component. Courses did not follow the convention and caused some confusion (see #18651).

This PR:
* Moves the wrapped component into the wrapping file.
* Moves a constant out of a function and into a constants file. Presumably the clutter around this large constant was the motivation for the unconventional split.
* Replaces an unnecessary function `mapDispatchToProps` with an object.

## Testing
1. Visit the affected page on this branch and production:
   * https://calypso.live/help/courses?branch=fix/i18n/courses-translate
   * https://wordpress.com/help/courses
1. Verify that the component behaves the same

## History

Wraps `client/me/help/help-courses/courses.jsx`, which uses `this.props.translate` in `localize()`.

Detected in #18651, ~blocker for #18590.~
